### PR TITLE
Enhancement with deprecated_argument_lookup

### PR DESCRIPTION
**Pull request by yongtang**
The tf.losses.cosine_distance deprecated dim and switched to axis. This fix adds the enhancement of using deprecated_argument_lookup, which is used in all other arguments deprecations.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by martinwicke — APPROVED** [review]
Thanks!
